src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -462,217 +420,98 @@ protected override void PreStart()
-                        PartitionKeyEscapeHelper.Escape(persistenceId)),
-                    TableOperators.And,
-                    TableQuery.GenerateFilterCondition(
-                        "RowKey",
-                        QueryComparisons.Equal,
-                        HighestSequenceNrEntry.RowKeyValue));
-
-            var returnValue = new TableQuery<HighestSequenceNrEntry>().Where(filter);
-
-            return returnValue;
+            return Table.QueryAsync<TableEntity>(
+                filter: $"PartitionKey eq '{PartitionKeyEscapeHelper.Escape(persistenceId)}' and " +
+                        $"RowKey eq '{HighestSequenceNrEntry.RowKeyValue}'",
+                maxPerPage: maxPerPage,
+                @select: null,
+                cancellationToken: cancellationToken
+            );
         }
 
-        //private static TableQuery GeneratePersistentJournalEntryDeleteQuery(
-        private static TableQuery<PersistentJournalEntry> GeneratePersistentJournalEntryDeleteQuery(
+        private AsyncPageable<TableEntity> PersistentJournalEntryDeleteQuery(
             string persistenceId,
-            long toSequenceNr)
+            long toSequenceNr,
+            int? maxPerPage = null,
+            CancellationToken cancellationToken = default)
         {
-            var persistenceIdFilter =
-                TableQuery.GenerateFilterCondition(
-                    "PartitionKey",
-                    QueryComparisons.Equal,
-                    PartitionKeyEscapeHelper.Escape(persistenceId));
-
-            var highestSequenceNrFilter =
-                TableQuery.GenerateFilterCondition(
-                    "RowKey",
-                    QueryComparisons.NotEqual,
-                    HighestSequenceNrEntry.RowKeyValue);
-
-            var rowKeyLessThanFilter =
-                TableQuery.GenerateFilterCondition(
-                    "RowKey",
-                    QueryComparisons.LessThanOrEqual,
-                    toSequenceNr.ToJournalRowKey());
-
-            var rowKeyFilter =
-                TableQuery.CombineFilters(
-                    highestSequenceNrFilter,
-                    TableOperators.And,
-                    rowKeyLessThanFilter);
-
-            var filter =
-                TableQuery.CombineFilters(
-                    persistenceIdFilter,
-                    TableOperators.And,
-                    rowKeyFilter);
-
-            var returnValue = new TableQuery<PersistentJournalEntry>().Where(filter);
-
-            return returnValue;
+            return Table.QueryAsync<TableEntity>(
+                filter: $"PartitionKey eq '{PartitionKeyEscapeHelper.Escape(persistenceId)}' and " +
+                        $"RowKey ne '{HighestSequenceNrEntry.RowKeyValue}' and " +
+                        $"RowKey le '{toSequenceNr.ToJournalRowKey()}'",
+                maxPerPage: maxPerPage,
+                @select: null,
+                cancellationToken: cancellationToken);
```

Comment:
Comparing this and the old way of building queries, you can see that this is a lot more clear and concise.